client/src/reducers/controls.js

```diff
@@ -75,6 +78,39 @@ const Controls = (
       const categoricalAsBooleansMap = createCategoricalAsBooleansMap(world);
       const crossfilter = Crossfilter(world.obsAnnotations);
       const dimensionMap = World.createObsDimensionMap(crossfilter, world);
+
+      let worldVarDataCache = world.varDataCache;
+
+      /* var dimensions */
+      if (userDefinedGenes.length > 0) {
+        /*
+          verbose & slightly confusing that we also access this as an object
+          in controls rather than an array, should be abstracted into
+          util ie., createDimensionsFromBothListsOfGenes(userGenes, diffExp)
+        */
+        _.forEach(userDefinedGenes, (gene, index) => {
+          dimensionMap[gene] = World.createVarDimension(
```

Comment:
dimensionMap keyed with a flat namespace that allows for collisions between users (obs annotations, var data names aka gene names, and layout X & Y)